core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -550,9 +550,16 @@ oppia.factory('explorationRightsService', [
         var that = this;
 
         requestParams.version = explorationData.data.version;
-        var explorationRightsUrl = (
-          '/createhandler/rights/' + explorationData.explorationId);
-        $http.put(explorationRightsUrl, requestParams).then(function(response) {
+        var requestUrl = '';
```

Comment:
No, I think you need to take this change a bit further -- separate concepts need separate functions. Declare new public functions here of the form publishExploration(), unpublishExploration(), publicizeExploration(), unpublicizeExploration(), etc. Pass in, explicitly, the values required for each, rather than bundling them in a requestParams() blob.